### PR TITLE
openPMD_api: Update to new version

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: O/openPMD_api/build_tarballs.jl [hunk 1]
```diff
@@ -121,7 +121,7 @@ dependencies = [
     Dependency(PackageSpec(name="ADIOS2_jll"); platforms=filter(p -> nbits(p) ≠ 32, platforms)),
     Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
     Dependency(PackageSpec(name="HDF5_jll"); compat="~1.14", platforms=hdf5_platforms),
-    Dependency(PackageSpec(name="libcxxwrap_julia_jll")),
+    Dependency(PackageSpec(name="libcxxwrap_julia_jll"); compat="~0.9"),
```
Comment (barche) [inline O/openPMD_api/build_tarballs.jl:124]: libcxxwrap 0.11 should be the correct one for CxxWrap 0.14, libcxxwrap 0.10 was yanked.

Comment (barche) [reply O/openPMD_api/build_tarballs.jl:124]: Yes, libcxxwrap 0.9.x is compatible with CxxWrap 0.13, libcxxwrap 0.11 is compatible with CxxWrap 0.14 (released yesterday).

**Comment by barche** [comment]
I made an issue for the 0.11 compile error: https://github.com/openPMD/openPMD-api/issues/1526

(in the wrong repo by accident 😬)

**Comment by franzpoeschel** [comment]
You're probably noticing a slight breaking change that we did on `openPMD_datatypes`. We had linker trouble with the constant and since it's a barely-used API, we replaced it with a function. 
So, yeah, the Julia bindings will need to adapt to this. (This situation is one of the reasons why I'm in favor of trying to go for a single repo in the end)

**Comment by eschnett** [comment]
Thanks for all the pointers. As it is difficult to match C++ and Julia datatypes (Julia is much more flexible), the Julia bindings manages most of the datatype wrangling by itself. Thus returning a copy of the `std::vector` from the C++ openPMD library isn't the right thing to do anyway.

File: O/openPMD_api/build_tarballs.jl [hunk 2]
```diff
@@ -73,8 +73,8 @@ cmake \
     -DBUILD_CLI_TOOLS=OFF \
     -DBUILD_EXAMPLES=OFF \
     -DBUILD_TESTING=OFF \
-    -DJulia_PREFIX=${prefix} \
-    -DopenPMD_USE_Julia=ON \
+    -DJULIA_PREFIX=${prefix} \
```
Comment (barche) [inline O/openPMD_api/build_tarballs.jl:76]: The `Julia_PREFIX` was the correct case here.

Comment (eschnett) [reply O/openPMD_api/build_tarballs.jl:76]: Yes it was. Thanks for spotting this!

**Comment by barche** [comment]
Looks good to me, I can merge this if it's OK for you @eschnett

**Comment by eschnett** [comment]
@barche Yes, this is ready.

**Comment by eschnett** [comment]
Yay me. I pushed the button, and I realized too late that the version number changed from `0.16.0` to `0.16.2`. What should I do?

**Comment by eschnett** [comment]
All is fine, the package was accepted anyway.

**Pull request merged**